### PR TITLE
Refactoring in the Pagination buttons

### PR DESCRIPTION
**Pull request by alexwlchan**
## Who is this for?

Devs.

## What is it doing for them?

Simplifying the code for the pagination buttons. I started looking at them for https://github.com/wellcomecollection/wellcomecollection.org/issues/7633, and kept tugging on the thread.

* Now you pass the total/current page and the pagination root, and the component figures out the mechanics of what the individual URLs should look like.
* I've added more examples to the Cardigan story, in particular when we're at the start/end of pagination.
* The catalogue and content apps were both using v similar implementations, so I've combined them into one component.

### CODE REVIEW

File: common/views/components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -98,64 +155,3 @@ const Pagination: FunctionComponent<Props> = ({
 };
 
 export default Pagination;
-export class PaginationFactory {
```
Comment (alexwlchan) [inline common/views/components/Pagination/Pagination.tsx:101]: This code isn't used anywhere, so I deleted it.

File: common/views/components/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
-  };
+};
+
+export type Props = {
+  paginatedResults: PaginatedResultsProps;
+  paginationRoot: LinkProps;
+  hideMobilePagination?: boolean;
+  disabled?: boolean;
+  onPageChange?: PageChangeFunction;
+  showPortal?: boolean;
```
Comment (alexwlchan) [inline common/views/components/Pagination/Pagination.tsx:23]: These extra parameters all come from the catalogue Paginator, but in the content app they're unused.

File: content/webapp/pages/events.tsx [hunk 3]
```diff
@@ -112,7 +112,7 @@ const EventsPage: FC<Props> = props => {
             },
           ]}
           paginatedResults={convertedPaginatedResults}
-          paginationRoot={`events${period ? `/${period}` : ''}`}
+          paginationRoot={`/events${period ? `/${period}` : ''}`}
```
Comment (alexwlchan) [inline content/webapp/pages/events.tsx:115]: I've double checked the pagination and canonical URLs; pushing this slash up here doesn't break anything.

**Review by rcantin-w — APPROVED** [review]
Works nicely locally too, thanks for doing that 🙌 🙌

**Pull request merged**